state/pruner.go

```diff
@@ -323,39 +313,31 @@ func (p *Pruner) pruneABCIResToRetainHeight(lastRetainHeight int64) int64 {
 	return newRetainHeight
 }
 
-// If no retain height has been set by the application or the data companion
-// the database will not have values for the corresponding keys.
-// If both retain heights were set, we pick the smaller one
-// If only one is set we return that one
-func (p *Pruner) findMinRetainHeight() int64 {
-	noAppRetainHeightSet := false
+func (p *Pruner) findMinBlockRetainHeight() int64 {
 	appRetainHeight, err := p.stateStore.GetApplicationRetainHeight()
 	if err != nil {
-		if !errors.Is(err, ErrKeyNotFound) {
-			p.logger.Error("Unexpected error fetching application retain height", "err", err)
-			return 0
-		}
-		noAppRetainHeightSet = true
+		p.logger.Error("Unexpected error fetching application retain height", "err", err)
+		return 0
+	}
+	// We only care about the companion retain height if pruning is configured
+	// to respect the companion's retain height.
+	if !p.dcEnabled {
+		return appRetainHeight
 	}
 	dcRetainHeight, err := p.stateStore.GetCompanionBlockRetainHeight()
 	if err != nil {
-		if !errors.Is(err, ErrKeyNotFound) {
```

Comment:
This again relies on the assumption the dc companion has to set at least the initial retain height in the initial config.

Reply:
Yup, that's the reasoning behind this PR. If they don't explicitly set it, it'll default to 0 as per Go's conventions, which is intended in this PR.

Reply:
Ok then, as it is 0 it also means nothing will be pruned until there is a higher retain height set.